client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -60,6 +62,39 @@ class StatsPostDetail extends Component {
 		page( '/stats/' + pathParts[ pathParts.length - 1 ] );
 	};
 
+	getNavigationItemsWithTitle = ( title ) => {
+		// TODO: Localize this.
+		// Currently not recognizing 'translate' as a valid function.
+		const localizedTabNames = {
+			traffic: 'Traffic',
+			insights: 'Insights',
+			store: 'Store',
+			ads: 'Ads',
+		};
+		const possibleBackLinks = {
+			traffic: '/stats/day/',
+			insights: '/stats/insights/',
+			store: '/stats/store/',
+			ads: '/stats/ads/',
+		};
+		// TODO: Don't use localStorage?
+		// This feels dirty but it works for now.
+		const lastClickedTab = localStorage.getItem( 'jp-stats-last-tab' );
+		const backLabel = localizedTabNames[ lastClickedTab ] || localizedTabNames.traffic;
+		let backLink = possibleBackLinks[ lastClickedTab ] || possibleBackLinks.traffic;
+		// Append the domain.
+		const domain = this.props?.path.split( '/' ).pop();
```

Comment:
```suggestion
		const domain = this.props.path?.split( '/' ).pop();
```

Component properties will never be falsey, but the path property can be.

Reply:
I believe the code as I wrote it is properly guarding the `path` access, not the `props` access. Your change would be guarding against the function `split()` being undefined. — [reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Let me know if i'm reading the docs incorrectly but that's how I understand it.

Reply:
You could try the following snippet in a JavaScript console:

```js
const props = { path: '/' };
console.log( '1', props.path?.split( '/' ) ); // split executes
console.log( '2', props?.path.split( '/' ) ); // split executes
console.log( '3', props.path2?.split( '/' ) ); // returns undefined, split doesn't execute
console.log( '4', props?.path2.split( '/' ) ); // throws error
```

`props` is never nullish, but `props.path2` can be nullish since objects return `undefined` when asked to retrieve a value for a non-existent key.

Reply:
Thanks for clarifying. I was doing some similar testing in the console yesterday but somehow I wasn't grokking the results correctly. I'll get these changes in today.